File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.c
int firstUniqChar(char* s) {
    int freq[26] = {0};
    int i;

    for (i = 0; s[i] != '\0'; i++){
        freq[s[i] - 'a']++;
    }

    for (i = 0; s[i] != '\0'; i++){
        if (freq[s[i] - 'a'] == 1){
            return i;
        }
    }
    return -1;
}

File: 0387-first-unique-character-in-a-string/0387-first-unique-character-in-a-string.py
class Solution:
    def firstUniqChar(self, s: str) -> int:
        hashmap = {}

        for ch in s:
            if ch in hashmap:
                hashmap[ch] = -1 #repeating character
            else:
                hashmap[ch] = 1
        
        for i in range(len(s)):
            if hashmap[s[i]] == 1:
                return i
        
        return -1